src/plugins/dataflow-tool/components/ui/dataflow-program-topbar.tsx

```diff
@@ -100,9 +104,46 @@ const RecordButton = (props: RecordButtonProps) => {
 };
 
 export const DataflowProgramTopbar = (props: TopbarProps) => {
+  const { serialDevice } = props;
+  const lastMsg = localStorage.getItem("last-connect-message");
+  const classes = classNames(
+    "icon-serial",
+    { "physical-connection": lastMsg === "connect"},
+    { "no-physical-connection": lastMsg === "disconnect"},
+    serialDevice.serialNodesCount > 0 ? "nodes-in-need" : "no-serial-needed",
+    serialDevice.hasPort() ? "has-port" : "no-port"
+  );
+
+  function serialMessage(){
+
+    // nodes that use serial, but no device physucally connected
```

Comment:
```suggestion
    // nodes that use serial, but no device physically connected
```

Reply:
good catch, thanks, will try to minimize these sorts of things!